Crashed when use platform_type in datasource ncloud_server_image

### Terraform CLI and Terraform Ncloud Provider Version

```text
Terraform version : v1.0.9
Ncloud Provider version : v2.1.2
```

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->
datasource "ncloud_server_image"

### Terraform Configuration Files

```hcl
data "ncloud_server_image" "test" {
  platform_type = "UBS64"
}
```

### Debug Output

Stack trace from the terraform-provider-ncloud_v2.1.2 plugin:

panic: interface conversion: interface {} is string, not []interface {}

goroutine 54 [running]:
github.com/terraform-providers/terraform-provider-ncloud/ncloud.getVpcServerImageProductList(0xc0000acf00, 0xc0003909c0, 0xc0007af810, 0x100eb6d, 0x203000, 0xc000694ea0, 0xc0007af8d8)
        github.com/terraform-providers/terraform-provider-ncloud/ncloud/data_source_ncloud_server_image.go:177 +0xbdf
github.com/terraform-providers/terraform-provider-ncloud/ncloud.getServerImageProductListFiltered(0xc0000acf00, 0xc0003909c0, 0xc0000acf00, 0x80, 0x80, 0x1d9ce00, 0x10780de)
        github.com/terraform-providers/terraform-provider-ncloud/ncloud/data_source_ncloud_server_image.go:104 +0x46
github.com/terraform-providers/terraform-provider-ncloud/ncloud.dataSourceNcloudServerImageRead(0xc0000acf00, 0x1be0fc0, 0xc0003909c0, 0x25b1920, 0xc000100000)
        github.com/terraform-providers/terraform-provider-ncloud/ncloud/data_source_ncloud_server_image.go:84 +0x4e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0xc0000d6f20, 0x1f8d628, 0xc0003023c0, 0xc0000acf00, 0x1be0fc0, 0xc0003909c0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.0.3/helper/schema/resource.go:281 +0x88
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).ReadDataApply(0xc0000d6f20, 0x1f8d628, 0xc0003023c0, 0xc0003e46a0, 0x1be0fc0, 0xc0003909c0, 0xc0003909c0, 0xc0003e46a0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.0.3/helper/schema/resource.go:489 +0xfd
github.com/hashicorp/terraform-plugin-sdk/v2/internal/helper/plugin.(*GRPCProviderServer).ReadDataSource(0xc00000c978, 0x1f8d628, 0xc0003023c0, 0xc000302340, 0xc00000c978, 0xc00000c988, 0x1e21c70)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.0.3/internal/helper/plugin/grpc_provider.go:1102 +0x4cc
github.com/hashicorp/terraform-plugin-sdk/v2/internal/tfplugin5._Provider_ReadDataSource_Handler.func1(0x1f8d628, 0xc0003023c0, 0x1d63920, 0xc000302340, 0xc0003023c0, 0x1cacfc0, 0xc000544001, 0xc00059c1b0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.0.3/internal/tfplugin5/tfplugin5.pb.go:3348 +0x89
github.com/hashicorp/terraform-plugin-sdk/v2/plugin.Serve.func3.1(0x1f8d6d0, 0xc000544000, 0x1d63920, 0xc000302340, 0xc0003e4240, 0xc00059c1b0, 0xc00031cba0, 0x109ae46, 0x1d33bc0, 0xc000544000)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.0.3/plugin/serve.go:76 +0x87
github.com/hashicorp/terraform-plugin-sdk/v2/internal/tfplugin5._Provider_ReadDataSource_Handler(0x1d96fe0, 0xc00000c978, 0x1f8d6d0, 0xc000544000, 0xc000194660, 0xc00000c5a0, 0x1f8d6d0, 0xc000544000, 0xc000588000, 0x124)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.0.3/internal/tfplugin5/tfplugin5.pb.go:3350 +0x150
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0000e96c0, 0x1f969f8, 0xc000196f00, 0xc000308200, 0xc000718510, 0x2573430, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.30.0/server.go:1171 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0000e96c0, 0x1f969f8, 0xc000196f00, 0xc000308200, 0x0)
        google.golang.org/grpc@v1.30.0/server.go:1494 +0xccf
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0000342e0, 0xc0000e96c0, 0x1f969f8, 0xc000196f00, 0xc000308200)
        google.golang.org/grpc@v1.30.0/server.go:834 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.30.0/server.go:832 +0x1fd

Error: The terraform-provider-ncloud_v2.1.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

### Steps to Reproduce

It's alway got error when I add `platform_type` to `ncloud_server_image`.

--- a/ncloud/data_source_ncloud_server_image.go
+++ b/ncloud/data_source_ncloud_server_image.go
@@ -2,6 +2,7 @@ package ncloud
 
 import (
 	"fmt"
+	"github.com/NaverCloudPlatform/ncloud-sdk-go-v2/ncloud"
 	"github.com/NaverCloudPlatform/ncloud-sdk-go-v2/services/server"
 	"github.com/NaverCloudPlatform/ncloud-sdk-go-v2/services/vserver"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
@@ -128,7 +129,7 @@ func getClassicServerImageProductList(d *schema.ResourceData, config *ProviderCo
 	}
 
 	if v, ok := d.GetOk("platform_type"); ok {
-		reqParams.PlatformTypeCodeList = expandStringInterfaceList(v.([]interface{}))
+		reqParams.PlatformTypeCodeList = []*string{ncloud.String(v.(string))}
 	}
 
 	logCommonRequest("GetServerImageProductList", reqParams)
@@ -174,7 +175,7 @@ func getVpcServerImageProductList(d *schema.ResourceData, config *ProviderConfig
 	}
 
 	if v, ok := d.GetOk("platform_type"); ok {
-		reqParams.PlatformTypeCodeList = expandStringInterfaceList(v.([]interface{}))
+		reqParams.PlatformTypeCodeList = []*string{ncloud.String(v.(string))}
 	}
 
 	logCommonRequest("GetServerImageProductList", reqParams)

--- a/ncloud/data_source_ncloud_server_image_test.go
+++ b/ncloud/data_source_ncloud_server_image_test.go
@@ -131,6 +131,30 @@ func testAccDataSourceNcloudServerImageByBlockStorageSize(t *testing.T, isVpc bo
 	})
 }
 
+func TestAccDataSourceNcloudServerImage_vpc_by_platform_type(t *testing.T) {
+	dataName := "data.ncloud_server_image.test5"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccDataSourceNcloudServerImageByPlatformTypeConfig("LNX64"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDataSourceID(dataName),
+					resource.TestCheckResourceAttr(dataName, "product_code", "SW.VSVR.APP.LNX64.CNTOS.0708.PINPT.LATEST.B050"),
+					resource.TestCheckResourceAttr(dataName, "product_name", "Pinpoint-centos-7.8-64"),
+					resource.TestCheckResourceAttr(dataName, "product_description", "CentOS 7.8 with Pinpoint"),
+					resource.TestCheckResourceAttr(dataName, "infra_resource_type", "SW"),
+					resource.TestCheckResourceAttr(dataName, "os_information", "CentOS 7.8 with Pinpoint"),
+					resource.TestCheckResourceAttr(dataName, "platform_type", "LNX64"),
+					resource.TestCheckResourceAttr(dataName, "base_block_storage_size", "50GB"),
+				),
+			},
+		},
+	})
+}
+
 func testAccDataSourceNcloudServerImageByCodeConfig(productCode string) string {
 	return fmt.Sprintf(`
 data "ncloud_server_image" "test1" {
@@ -172,3 +196,12 @@ data "ncloud_server_image" "test4" {
 	}
 }
 `
+
+func testAccDataSourceNcloudServerImageByPlatformTypeConfig(platformType string) string {
+	return fmt.Sprintf(`
+data "ncloud_server_image" "test5" {
+  product_code = "SW.VSVR.APP.LNX64.CNTOS.0708.PINPT.LATEST.B050"
+  platform_type = "%s"
+}
+`, platformType)
+}

--- a/ncloud/data_source_ncloud_server_image_test.go
+++ b/ncloud/data_source_ncloud_server_image_test.go
@@ -131,7 +131,7 @@ func testAccDataSourceNcloudServerImageByBlockStorageSize(t *testing.T, isVpc bo
 	})
 }
 
-func TestAccDataSourceNcloudServerImage_vpc_by_platform_type(t *testing.T) {
+func TestAccDataSourceNcloudServerImage_vpc_byPlatformType(t *testing.T) {
 	dataName := "data.ncloud_server_image.test5"
 
 	resource.ParallelTest(t, resource.TestCase{

